File: 프로그래머스/LV2/코딩테스트 연습/연습문제/JadenCase 문자열 만들기/JadenCase 문자열 만들기.cpp
#include <string>
#include <vector>

using namespace std;

string answer = "";

void go(char before, string s, int s_idx, int s_len) {
    if(s_idx == s_len)return;

    if(isdigit(before))answer += tolower(s[s_idx]);
    else if(before == ' ')answer += toupper(s[s_idx]);
    else answer += tolower(s[s_idx]);

    go(s[s_idx], s, s_idx+1, s_len);

}

string solution(string s) {
    answer = isalpha(s[0]) ? toupper(s[0]) : s[0];
    go(s[0], s, 1, s.length());
    return answer;
}

File: 프로그래머스/LV2/코딩테스트 연습/연습문제/JadenCase 문자열 만들기/JadenCase 문자열 만들기.py
answer = ""

def go(before, s, s_idx, s_len):
    global answer

    if s_idx == s_len:
        return

    if before.isdigit():
        answer += s[s_idx].lower()
    elif before == ' ':
        answer += s[s_idx].upper()
    else:
        answer += s[s_idx].lower()

    go(s[s_idx], s, s_idx+1, s_len)

def solution(s):
    global answer
    answer = s[0].upper() if s[0].isalpha() else s[0]
    go(s[0], s, 1, len(s))
    return answer